Source/MMWormholeManifestFileTransiting.m

```diff
@@ -0,0 +1,140 @@
+
+- (BOOL)writeMessageObject:(id<NSCoding>)messageObject forIdentifier:(NSString *)identifier {
+    if (identifier == nil) {
+        return NO;
+    }
+    
+    if (messageObject) {
+        //Insert into manifest
+        NSMutableArray *storageArray = nil;
+        NSArray *potentialOnDiskArray = (NSArray *)[self _arrayForIdentifier:identifier];
+        if (potentialOnDiskArray) {
+            storageArray = [NSMutableArray arrayWithArray:potentialOnDiskArray];
+        } else {
+            storageArray = [[NSMutableArray alloc] init];
+        }
+        [storageArray addObject:messageObject];
+        
+        //Encode manifest instead of message object
+        BOOL success = [self _writeArrayToDisk:storageArray forIdentifier:identifier];
+        
+        if (!success) {
+            return NO;
+        }
+    }
+    
+    return YES;
+}
+
+- (id<NSCoding>)messageObjectForIdentifier:(NSString *)identifier {
+    if (identifier == nil) {
+        return nil;
+    }
+    
+    NSArray *storageArray = [self _arrayForIdentifier:identifier];
+    if ([storageArray count] == 0) {
+        return nil;
+    }
+    id messageObject = storageArray[0];
+    NSMutableArray *arrayToWrite = [NSMutableArray arrayWithArray:storageArray];
+    [arrayToWrite removeObjectAtIndex:0];
+    
+    //TODO: What to do if this fails?
+    [self _writeArrayToDisk:arrayToWrite forIdentifier:identifier];
+    
+    return messageObject;
+}
+
+#pragma mark - Helper Methods
+
+- (BOOL)_writeArrayToDisk:(NSArray *)array forIdentifier:(NSString *)identifier{
+    NSData *data = [NSKeyedArchiver archivedDataWithRootObject:array];
+    NSString *filePath = [self filePathForIdentifier:identifier];
+    NSURL *fileURL = [NSURL fileURLWithPath:filePath];
+    
+    if (data == nil || filePath == nil || fileURL == nil) {
+        return NO;
+    }
+    
+    NSFileCoordinator *fileCoordinator = [[NSFileCoordinator alloc] initWithFilePresenter:nil];
+    NSError *error = nil;
+    __block BOOL success = NO;
+    
+    [fileCoordinator
+     coordinateWritingItemAtURL:fileURL
+     options:0
+     error:&error
+     byAccessor:^(NSURL *newURL) {
+         NSError *writeError = nil;
+         
+         success = [data writeToURL:newURL
+                            options:NSDataWritingAtomic | self.additionalFileWritingOptions
+                              error:&writeError];
+     }];
+    
+    return success;
+}
+
+- (nullable NSArray*)_arrayForIdentifier:(NSString *)identifier{
+    if (identifier == nil) {
+        return nil;
+    }
+    
+    NSString *filePath = [self filePathForIdentifier:identifier];
+    NSURL *fileURL = [NSURL fileURLWithPath:filePath];
+    
+    if (filePath == nil || fileURL == nil) {
+        return nil;
+    }
+    
+    NSFileCoordinator *fileCoordinator = [[NSFileCoordinator alloc] initWithFilePresenter:nil];
+    NSError *error = nil;
+    __block NSData *data = nil;
+    
+    [fileCoordinator
```

Comment:
Seems expensive to read from disk for every call, though I'd probably just make a note of the consideration until it's a measurable issue.